assets/js/payment-methods-demo/express-payment/apple-pay/apple-pay-express.js

```diff
@@ -0,0 +1,380 @@
+				onShippingSelectedRate()
+			);
+			const unsubscribeShippingRateSelectFail = subscriber.onShippingRateSelectFail(
+				onShippingRatesEvent( false )
+			);
+			const unsubscribeCheckoutCompleteSuccess = subscriber.onCheckoutCompleteSuccess(
+				onCheckoutComplete()
+			);
+			const unsubscribeCheckoutCompleteFail = subscriber.onCheckoutCompleteError(
+				onCheckoutComplete( false )
+			);
+			return () => {
+				unsubscribeCheckoutCompleteFail();
+				unsubscribeCheckoutCompleteSuccess();
+				unsubscribeShippingRateFail();
+				unsubscribeShippingRateSuccess();
+				unsubscribeShippingRateSelectSuccess();
+				unsubscribeShippingRateSelectFail();
+			};
+		}
+	}, [ canMakePayment ] );
+
+	// locale is not a valid value for the paymentRequestButton style
+	const { theme } = getStripeServerData().button;
+
+	const paymentRequestButtonStyle = {
+		paymentRequestButton: {
+			type: 'default',
+			theme,
+			height: '48px',
+		},
+	};
+
+	return canMakePayment && paymentRequest ? (
+		<PaymentRequestButtonElement
+			onClick={ onButtonClick }
+			options={ {
+				style: paymentRequestButtonStyle,
+				paymentRequest,
+			} }
+		/>
+	) : null;
+};
+
+/**
+ * ApplePayExpress with stripe provider
+ *
+ * @param {RegisteredPaymentMethodProps|{}} props
+ */
+export const ApplePayExpress = ( props ) => {
+	const { locale } = getStripeServerData().button;
+	return (
+		<Elements stripe={ stripePromise } locale={ locale }>
+			<ApplePayExpressComponent { ...props } />
+		</Elements>
+	);
+};
+
+/**
+ * Notes for `stripe-payment-request`:
+ *
+ * - startPaymentRequest appears to be where the payment request button is setup,
+ *   determination of whether the payment can be made and adding event handlers.
+ * - much of this logic will be provided from the checkout/cart so the payment
+ *   method handler here will just have to implement things like canMakePayment
+ *   and expose the button with the appropriate onClick handler.
+ * - this also utilizes paymentRequestType which comes from the `canMakePayment`
+ *   call.
+ * - because this is an apple pay express payment, we need to handle if apple_pay
+ *  is not available.
+ *
+ * Note payment scripts are enqueued in `class-wc-stripe-payment-request` and that
+ * includes enqueuing version 3 of the stripe js client (which is what is being
+ * used for stripe payments in the stripe plugin)
+ */
+
+// activeContent receives:
+// - checkoutData (which comes from useCheckoutData)
+// - paymentEvents (which is an object of `dispatch` and `select` from the usePaymentEvents hook)
+
+//@todo suggested endpoint for the checkout to do it's processing on submit:
+/**
+/store/checkout/payment/{order_id}
+body: {
+  paymentMethod: 'applePay',
+  // arbitrary things the payment needs to process - eg. token, payment.source.order_id
+  // that have no totals or anything and are simply used for processing the
+  // payment server side.
+}
+ */
```

Comment:
Nothing blocking here yet but there is a lot of code and events to track for express methods - maybe we can reduce whats needed when we do some more express methods and see what gets duplicated? Before it's made public I mean.

Reply:
Agreed